### PR TITLE
Plugin update manager: empty state when no eligible plugins are installed

### LINKED ISSUE #88404: FE: provide empty state for plugin picker
In the create/edit schedule, let's provide an empty state for the plugin list view.

cc @JanaMW27 

Context: https://github.com/Automattic/wp-calypso/pull/88376#issuecomment-1988970849

### PR DESCRIPTION
**Pull request by TimBroddin**
Closes https://github.com/Automattic/wp-calypso/issues/88404

## Proposed Changes

- Checks whether there are eligble plugins installed inside `<PluginsUpdateManager /> ` and sets this on the context
- Shows a message + CTA on the list screen informing the user that they have no eligble plugins
- Minor refactorings

CleanShot 2024-03-18 at 13 41 38@2x

## Testing Instructions

1. Apply this PR
2. Check whether you see the message when no eligble plugins are installed (you can mock this by returning `[]` inside the select function of `client/data/plugins/use-core-plugins-query.ts`.
3. Tets for regressions

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by JanaMW27** [comment]
Nice one!

A couple of thoughts:
- Let's remove the + from the CTA
- In the sentence I would replace "manage" with "update" since this section is all about update schedules rather than managing the plugin per se.

File: client/blocks/plugins-update-manager/index.tsx [hunk 1]
```diff
@@ -40,6 +41,14 @@ export const PluginsUpdateManager = ( props: Props ) => {
 	const hideCreateButton =
 		! isEligibleForFeature || schedules.length === MAX_SCHEDULES || schedules.length === 0;
 
+	const { data: plugins = [], isFetched: isPluginsFetched } = useCorePluginsQuery(
+		siteSlug,
+		true,
+		true
+	);
+	const siteHasEligiblePlugins =
+		isPluginsFetched && plugins.filter( ( plugin ) => ! plugin.is_managed ).length > 0;
```
Comment (bogiii) [inline client/blocks/plugins-update-manager/index.tsx]: This filtering has already been done using `useCorePluginsQuery`; see the second `true` parameter. It is for filtering managed plugins.

**Comment by TimBroddin** [comment]
> Nice one!
> 
> A couple of thoughts:
> 
> * Let's remove the + from the CTA
> * In the sentence I would replace "manage" with "update" since this section is all about update schedules rather than managing the plugin per see.

I removed the + and changed the copy. Thanks for the review @JanaMW27 

CleanShot 2024-03-18 at 14 41 43@2x

File: client/blocks/plugins-update-manager/hooks/use-site-has-eligible-plugins.ts [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
-		}
-	}, [ data, isFetched ] );
-}
-
 export function useSiteHasEligiblePlugins() {
-	const { siteHasEligiblePlugins } = useContext( PluginUpdateManagerContext );
-	return siteHasEligiblePlugins;
+	const { siteHasEligiblePlugins, siteHasEligiblePluginsLoading } = useContext(
+		PluginUpdateManagerContext
+	);
```
Comment (bogiii) [inline client/blocks/plugins-update-manager/hooks/use-site-has-eligible-plugins.ts:12]: From my point of view, it looks like an overhead setting and getting this data from the context.

We can use it directly from the `useCorePluginsQuery`; it persists and shares the info between components, so we can have the right data if it's loading and the plugin data.

What do you think?

Comment (TimBroddin) [reply client/blocks/plugins-update-manager/hooks/use-site-has-eligible-plugins.ts:12]: > From my point of view, it looks like an overhead setting and getting this data from the context.
> 
> We can use it directly from the `useCorePluginsQuery`; it persists and shares the info between components, so we can have the right data if it's loading and the plugin data.
> 
> What do you think?

I refactored the logic:
- Moved all the logic to `useSiteHasEligiblePlugins` and removed setting it on the context
- Allow passing the siteSlug to `useSiteHasEligiblePlugins` for usage inside the root `<PluginsUpdateManager />` component

File: client/blocks/plugins-update-manager/schedule-list.tsx [hunk 3]
```diff
@@ -52,8 +55,15 @@ export const ScheduleList = ( props: Props ) => {
 		isEligibleForFeature
 	);
 
+	const isLoading =
+		isLoadingSchedules ||
+		isLoadingCanCreateSchedules ||
+		isEligibleForFeatureLoading ||
+		siteHasEligiblePluginsLoading;
```
Comment (bogiii) [inline client/blocks/plugins-update-manager/schedule-list.tsx:62]: I like it; it's more readable. 👍

File: client/blocks/plugins-update-manager/hooks/use-site-has-eligible-plugins.ts (hunk 2 above)
Comment (bogiii) [reply client/blocks/plugins-update-manager/hooks/use-site-has-eligible-plugins.ts:12]: Thanks for handling it. Nice work. 👍

**Review by bogiii — APPROVED** [review]
Nice work! No regression were found. 🚢

**Pull request merged**